network/hnswrapper/hnsv2wrapperwithtimeout.go

```diff
@@ -0,0 +1,388 @@
+// Copyright 2022 Microsoft. All rights reserved.
+// MIT License
+
+//go:build windows
+// +build windows
+
+package hnswrapper
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Microsoft/hcsshim/hcn"
+)
+
+const (
+	// hnsCallTimeout indicates the time in seconds to wait for hns calls before timing out
+	hnsCallTimeout = 15 * time.Second
+)
+
+// ErrHNSCallTimeout - hns call timeout
+var ErrHNSCallTimeout = errors.New("timed out calling hns")
+
+type Hnsv2wrapperwithtimeout struct {
+	Hnsv2 HnsV2WrapperInterface
+}
+
+type CreateEndpointFuncResult struct {
+	endpoint *hcn.HostComputeEndpoint
+	Err      error
+}
+
+type GetEndpointByIDFuncResult struct {
+	endpoint *hcn.HostComputeEndpoint
+	Err      error
+}
+
+type ListEndpointsFuncResult struct {
+	endpoints []hcn.HostComputeEndpoint
+	Err       error
+}
+
+type CreateNetworkFuncResult struct {
+	network *hcn.HostComputeNetwork
+	Err     error
+}
+
+type GetNamespaceByIDFuncResult struct {
+	namespace *hcn.HostComputeNamespace
+	Err       error
+}
+
+type GetNetworkByNameFuncResult struct {
+	network *hcn.HostComputeNetwork
+	Err     error
+}
+
+type GetNetworkByIDFuncResult struct {
+	network *hcn.HostComputeNetwork
+	Err     error
+}
+
+func (h Hnsv2wrapperwithtimeout) CreateEndpoint(endpoint *hcn.HostComputeEndpoint) (*hcn.HostComputeEndpoint, error) {
+	r := make(chan CreateEndpointFuncResult)
+	ctx, cancel := context.WithTimeout(context.TODO(), hnsCallTimeout)
```

Comment:
can hnsCallTimeout be part of Hnsv2wrapperwithtimeout structure? Also do you want to follow camelcase notation for this structure name?

Reply:
I updated it to make it part of Hnsv2wrapperwithtimeout struct